src/kili/services/json_response/category.py

```diff
@@ -0,0 +1,112 @@
+# Feature is still under development and is not yet suitable for use by general users.
+"""Module for the "categories" key parsing of a job response."""
+from typing import Dict, List
+
+from typeguard import typechecked
+
+from .exceptions import InvalidMutationError
+
+
+class Category(Dict):
+    """Class for Category parsing."""
+
+    def __init__(self, job_interface: Dict, name: str, confidence: int = 100) -> None:
+        """Class for Category parsing.
+
+        Args:
+            name: Name of the category label.
+            confidence: Confidence of the category label.
+            job_interface: Job interface of the job.
+        """
+        super().__init__()
+        self.job_interface = job_interface
+
+        # call the setters
+        self.name = name
+        self.confidence = confidence
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the category label."""
+        return self["name"]
+
+    @name.setter
+    @typechecked
```

Comment:
wondering if we should not put `@typechecked` for all the public methods automatically

Reply:
in category.py there are only two setters I think

But I might add typechecked for all setters in the other PR for label mutations yes